### PR TITLE
sig-cluster-lifecycle: add annual report for 2023

### PR DESCRIPTION
**Pull request by neolit123**
**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/community/issues/7769

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
/cc @fabriziopandini @justinsb @vincepri 
/assign @palnabarun

**Review by vincepri — APPROVED** [review]

**Comment by justinsb** [comment]
Thanks for doing this @neolit123 ; lgtm!

File: sig-cluster-lifecycle/annual-report-2023.md [hunk 1]
```diff
@@ -20,16 +34,45 @@
...
 3. Did you have community-wide updates in 2023 (e.g. KubeCon talks)?
 
-<!--
-  Examples include links to email, slides, or recordings.
--->
+For KubeCon EU 2023, we had the following talks:
+- Minikube, from CLI to GUI
+- How to Turn Release Management from Duty to Fun: Lessons Learned Building the Cluster API Release Team
+- Cluster API Providers: Intro, Deep Dive, and Community!
+- Kubeadm Deep Dive
```
Comment (pacoxu) [inline sig-cluster-lifecycle/annual-report-2023.md:60]: Kops maintainer track:
https://kccnceu2023.sched.com/event/1HycF/mission-accomplished-kubernetes-is-not-a-monorepo-now-our-work-begins-justin-santa-barbara-google-ciprian-hacman-microsoft

**Review by fabriziopandini — COMMENTED** [review]
Thanks for writing this doc @neolit123!
/lgtm

File: sig-cluster-lifecycle/annual-report-2023.md (hunk 1 above)
Comment (neolit123) [reply sig-cluster-lifecycle/annual-report-2023.md:60]: added. it's more of a general history talk, but it's from the kOps maintainers.

**Comment by neolit123** [comment]
removed DRAFT state from PR description.
let's see if anything else can be added according to reviews, but we already have two LGTMs from leads above.

File: sig-cluster-lifecycle/annual-report-2023.md [hunk 2]
```diff
@@ -20,16 +34,46 @@
...
+- Mission Accomplished: Kubernetes Is Not a Monorepo. Now Our Work Begins!
 
-4. KEP work in 2023 (v1.27, v1.28, v1.29):
+For KubeCon NA 2023, we had the following talks:
+- SIG Cluster Lifecycle Intro & Future
+- Minikube project update
+- Cluster API Deep dive
```
Comment (palnabarun) [inline sig-cluster-lifecycle/annual-report-2023.md:66]: Could you please include links to these talks? Sched links should work.

File: sig-cluster-lifecycle/annual-report-2023.md [hunk 3]
```diff
@@ -20,16 +34,46 @@
...
+We encourage contributors who are interested in our subprojects to reach out to the SIG mailing list or to
+individual subproject Slack channels.
+
+
 3. Did you have community-wide updates in 2023 (e.g. KubeCon talks)?
 
-<!--
-  Examples include links to email, slides, or recordings.
--->
+For KubeCon EU 2023, we had the following talks:
```
Comment (palnabarun) [inline sig-cluster-lifecycle/annual-report-2023.md:56]: Could you please include links to these talks? Sched links should work.

File: sig-cluster-lifecycle/annual-report-2023.md [hunk 4]
```diff
@@ -20,16 +34,46 @@
       Please raise an issue in kubernetes/community, if the KEP metadata is correct but the generated list is incorrect.
 -->
 
+As established by our self-assessment mentioned above, the most common areas where we need help are:
+- Contributing to code
+- Reviewing code
```
Comment (palnabarun) [inline sig-cluster-lifecycle/annual-report-2023.md:39]: Are there any specific onboarding guides or list of issues/PRs for new contributors?

File: sig-cluster-lifecycle/annual-report-2023.md (hunk 3 above)
Comment (neolit123) [reply sig-cluster-lifecycle/annual-report-2023.md:56]: updated.

File: sig-cluster-lifecycle/annual-report-2023.md (hunk 2 above)
Comment (neolit123) [reply sig-cluster-lifecycle/annual-report-2023.md]: updated.

File: sig-cluster-lifecycle/annual-report-2023.md (hunk 4 above)
Comment (neolit123) [reply sig-cluster-lifecycle/annual-report-2023.md:39]: all the listed subprojects have the contributing.md, but only kubeadm has some extra info in it for onboarding new contributors like instructional videos.  it's up to the subprojects to understand that if they need contributors, they need to add better instructions. as a governance unit we can only suggest to them how to improve that.

also, in reality it feels like the quality of contributor onboarding facilities are not a big factor. if someone has the motivation (e.g. being paid)  to contribute they will reach out and find a way.

Comment (palnabarun) [reply sig-cluster-lifecycle/annual-report-2023.md:39]: > kubeadm has some extra info in it for onboarding new contributors like instructional videos

Amazing. 👍🏽 

> also, in reality it feels like the quality of contributor onboarding facilities are not a big factor. if someone has the motivation (e.g. being paid) to contribute they will reach out and find a way.

The last sentence is very true. But in addition, onboarding guides help in easing the friction and barriers of contributions and that's why we should have those. Think what if Kubernetes did not contributor onboarding documentation in k/community, maintainers would have needed to answer the same questions many many times more.

**Review by palnabarun — APPROVED** [review]
/lgtm

**Comment by neolit123** [comment]
@palnabarun drop hold on your own leisure, please.
SIG leads already LGTMed, previously.

**Pull request merged**